src/PricingManager/Rule/Dao.php

```diff
@@ -76,11 +76,19 @@ public function create(): void
      */
     public function save(): void
     {
-        if (!$this->model->getId()) {
-            $this->create();
+        try {
+            if (!$this->model->getId()) {
+                $this->create();
+            }
+
+            $this->update();
+            $this->db->commit();
+        } catch (\Exception $e) {
+            $this->db->rollBack();
+
+            throw $e;
         }
```

Comment:
Shouldn't you start the transaction first?

Reply:
```suggestion
        $this->db->beginTransaction();
        try {
            if (!$this->model->getId()) {
                $this->create();
            }

            $this->update();
            $this->db->commit();
        } catch (\Exception $e) {
            $this->db->rollBack();

            throw $e;
        }
```

Reply:
Maybe it's easier to use `transactional()` instead?

```suggestion
        $this->db->transactional(function () {
            if (!$this->model->getId()) {
                $this->create();
            }

            $this->update();
        });
```